west: spdx: Fix copyright parsing

sorru94 opened the pull request:
Fix the copyright parsing bug resulting from the new REUSE API. See the related issue for more information on the API change from v0.5.x and v0.6.x in REUSE.

Fixes #98378

marc-hb reviewed (commented):
I see you have a bug reference which is great but a brief summary of the issue in the commit message itself would not hurt. If you could also make the commit subject a bit more specific, even better! Admittedly very difficult in just one line and so few characters.

sorru94 commented:
> I see you have a bug reference which is great but a brief summary of the issue in the commit message itself would not hurt. If you could also make the commit subject a bit more specific, even better! Admittedly very difficult in just one line and so few characters.

Not sure if it's a good idea, but what do you think about squashing the two commits together?

sorru94 committed:
west: spdx: Fix copyright parsing in REUSE 0.6.x

Fix broken copyright parsing. Since REUSE 6.0.0 the copyright_lines
attribute has been removed. The new attribute copyright_notices is used.
Additionally, the python package REUSE should now be at least v0.6.0.
This will require an update of REUSE for users that were using a
version earlier than 0.6.

Fixes #98378

Signed-off-by: Simone Orru <simone.orru@secomind.com>

sorru94 commented:
@kartben No, the `CopyrightNotice` class has been added in 6.0
I rebased and pushed the single commit form.
